### PR TITLE
Wip/hughsie/cleanup

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

File: src/xb-builder.c [hunk 1]
```diff
@@ -435,7 +435,7 @@ static void
 xb_builder_nodetab_write_sentinel (XbBuilderNodetabHelper *helper)
 {
 	XbSiloNode sn = {
-		.is_node	= FALSE,
+		.flags		= XB_SILO_NODE_FLAG_NONE,
```
Comment (pwithnall) [inline src/xb-builder.c:438]: Nitpick: Would have been a bit easier to review this if the changes to add accessor functions were split out into separate commits, but it’s too late now.

File: src/xb-silo-export.c [hunk 2]
```diff
@@ -38,22 +38,22 @@ xb_silo_export_node (XbSilo *self, XbSiloExportHelper *helper, XbSiloNode *sn, G
 				xb_silo_from_strtab (self, sn->element_name));
 
 	/* add any attributes */
-	for (guint8 i = 0; i < sn->nr_attrs; i++) {
-		XbSiloNodeAttr *a = xb_silo_get_attr (self, helper->off, i);
+	for (guint8 i = 0; i < xb_silo_node_get_nr_attrs (sn); i++) {
```
Comment (pwithnall) [inline src/xb-silo-export.c:41]: Nitpick: `xb_silo_node_get_n_attrs()` or `xb_silo_node_get_attr_count()` would be more conventional ways of naming this. There are no examples of `*_get_nr_*` function names in devhelp for me.

File: src/xb-silo-export.c [hunk 3]
```diff
@@ -64,7 +64,7 @@ xb_silo_export_node (XbSilo *self, XbSiloExportHelper *helper, XbSiloNode *sn, G
 		helper->off += xb_silo_node_get_size (sn);
 
 		/* recurse deeper */
-		while (xb_silo_get_node(self, helper->off)->is_node) {
+		while (xb_silo_get_node(self, helper->off)->flags & XB_SILO_NODE_FLAG_IS_ELEMENT) {
```
Comment (pwithnall) [inline src/xb-silo-export.c:67]: Nitpick: You could fix the spacing after `xb_silo_get_node` while modifying this line

File: src/xb-silo-node.c [hunk 4]
```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
```
Comment (pwithnall) [inline src/xb-silo-node.c:2]: 2021?

File: src/xb-silo-node.c [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+#include <string.h>
+
+#include "xb-silo-node.h"
+
+/* private */
+guint32
+xb_silo_node_get_size (XbSiloNode *self)
+{
+	/* sentinal */
```
Comment (pwithnall) [inline src/xb-silo-node.c:17]: ‘sentinel’

File: src/xb-silo-node.h [hunk 6]
```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
```
Comment (pwithnall) [inline src/xb-silo-node.h:2]: 2021?

**Review by pwithnall — COMMENTED** [review]
Nitpick: Might be best [not to use the word ‘sanity’ in a negative context](https://text.npr.org/739643765) in the commit message. Perhaps ‘consistency’ or ‘readability’ would work?

File: src/xb-silo-node.h (hunk 6 above)
Comment (hughsie) [reply src/xb-silo-node.h:2]: Ta.

File: src/xb-silo-node.c (hunk 4 above)
Comment (hughsie) [reply src/xb-silo-node.c:2]: Ta.

File: src/xb-silo-node.c (hunk 5 above)
Comment (hughsie) [reply src/xb-silo-node.c:17]: Ta.

File: src/xb-silo-export.c (hunk 2 above)
Comment (hughsie) [reply src/xb-silo-export.c:41]: xb_silo_node_get_attr_count works much better, thanks!

**Comment by hughsie** [comment]
@pwithnall looks good to you now?

**Review by pwithnall — APPROVED** [review]
